File: leetcode/3001-4000/3042-count-prefix-and-suffix-pairs-i/CountPrefixAndSuffixPairsI.ts
function countPrefixSuffixPairs(words: string[]): number {
  let totalPrefixSuffixParis = 0;

  for (let i = 0; i < words.length; i++) {
    for (let j = i + 1; j < words.length; j++) {
      if (words[j].startsWith(words[i]) && words[j].endsWith(words[i])) {
        totalPrefixSuffixParis++;
      }
    }
  }

  return totalPrefixSuffixParis;
};

console.log(countPrefixSuffixPairs(["a", "aba", "ababa", "aa"]));
console.log(countPrefixSuffixPairs(["abab", "ab"]));

File: leetcode/3001-4000/3001-3100/3041-3050/3042-count-prefix-and-suffix-pairs-i/CountPrefixAndSuffixPairsI.java
public class CountPrefixAndSuffixPairsI {
  public static void main(String[] args) {
    // String[] words = { "a", "aba", "ababa", "aa" };
    String[] words = { "abab", "ab" };

    System.out.println(new Solution().countPrefixSuffixPairs(words));
  }
}

class Solution {
  public int countPrefixSuffixPairs(String[] words) {
    int totalPrevfixSuffixPairs = 0;

    for (int i = 0; i < words.length; i++) {
      for (int j = i + 1; j < words.length; j++) {
        if (words[j].startsWith(words[i]) && words[j].endsWith(words[i])) {
          totalPrevfixSuffixPairs++;
        }
      }
    }

    return totalPrevfixSuffixPairs;
  }
}
